gateway/src/apicast/policy/routing/README.md

```diff
@@ -0,0 +1,316 @@
+- [**Rule that uses a jwt claim**](#rule-that-uses-a-jwt-claim)
+- [**Rule with several operations**](#rule-with-several-operations)
+- [**Combining rules**](#combining-rules)
+- [**Supported operations**](#supported-operations)
+- [**Liquid templating**](#liquid-templating)
+
+## Description
+
+This policy allows to modify the upstream (scheme, host, and port) of a request
+based on:
+
+- The request path
+- A header
+- A query argument
+- A jwt claim
+
+## Rule that uses the request path
+
+This is a configuration that routes to `http://example.com` when the path is
+`/accounts`:
+
+```json
+  {
+    "name": "routing",
+    "version": "builtin",
+    "configuration": {
+      "rules": [
+        {
+          "url": "http://example.com",
+          "condition": {
+            "operations": [
+              {
+                "thing_to_match": "path",
+                "op": "==",
+                "value": "/accounts"
+              }
+            ]
+          }
+        }
+      ]
+    }
+  }
+```
+
+## Rule that uses a header
+
+This is a configuration that routes to `http://example.com` when the value of
+the header `Test-Header` is `123`:
+
+```json
+  {
+    "name": "routing",
+    "version": "builtin",
+    "configuration": {
+      "rules": [
+        {
+          "url": "http://example.com",
+          "condition": {
+            "operations": [
+              {
+                "thing_to_match": "header",
+                "header_name": "Test-Header",
+                "op": "==",
+                "value": "123"
+              }
+            ]
+          }
+        }
+      ]
+    }
+  }
+```
+
+## Rule that uses a query argument
+
+This is a configuration that routes to `http://example.com` when the value of
+the query argument `test_query_arg` is `123`:
+
+```json
+  {
+    "name": "routing",
+    "version": "builtin",
+    "configuration": {
+      "rules": [
+        {
+          "url": "http://example.com",
+          "condition": {
+            "operations": [
+              {
+                "thing_to_match": "query_arg",
+                "query_arg_name": "test_query_arg",
+                "op": "==",
+                "value": "123"
+              }
+            ]
+          }
+        }
+      ]
+    }
+  }
+```
+
+## Rule that uses a jwt claim
+
+In order to be able to route based on the value of a jwt claim, there needs to
+be a policy in the chain that validates the jwt and stores it in the context
+that the policies share.
+
+This is a configuration that routes to `http://example.com` when the value of
+the jwt claim `test_claim` is `123`:
+
+```json
+  {
+    "name": "routing",
+    "version": "builtin",
+    "configuration": {
+      "rules": [
+        {
+          "url": "http://example.com",
+          "condition": {
+            "operations": [
+              {
+                "thing_to_match": "jwt_claim",
+                "jwt_claim_name": "test_claim",
+                "op": "==",
```

Comment:
We should consider array-like claims, for example, "scopes" claim that might need "includes" and "excludes" operation.